tests/Single/ChunkwiseTest.php

```diff
@@ -0,0 +1,651 @@
+                3,
+                [[1]],
+            ],
+            [
+                $trav([1, 2]),
+                1,
+                [[1], [2]],
+            ],
+            [
+                $trav([1, 2]),
+                2,
+                [[1, 2]],
+            ],
+            [
+                $trav([1, 2]),
+                3,
+                [[1, 2]],
+            ],
+            [
+                $trav([1, 2, 3]),
+                1,
+                [[1], [2], [3]],
+            ],
+            [
+                $trav([1, 2, 3]),
+                2,
+                [[1, 2], [3]],
+            ],
+            [
+                $trav([1, 2, 3]),
+                3,
+                [[1, 2, 3]],
+            ],
+            [
+                $trav([1, 2, 3]),
+                4,
+                [[1, 2, 3]],
+            ],
+            [
+                $trav([1, 2, 3, 4]),
+                2,
+                [[1, 2], [3, 4]],
+            ],
+            [
+                $trav([1, 2, 3, 4]),
+                3,
+                [[1, 2, 3], [4]],
+            ],
+            [
+                $trav([1, 2, 3, 4, 5]),
+                2,
+                [[1, 2], [3, 4], [5]],
+            ],
+            [
+                $trav([1, 2, 3, 4, 5]),
+                3,
+                [[1, 2, 3], [4, 5]],
+            ],
+            [
+                $trav([1.1, 2.2, 3.3, 4.4, 5.5]),
+                2,
+                [[1.1, 2.2], [3.3, 4.4], [5.5]],
+            ],
+            [
+                $trav(['1', '2', '3', '4', '5']),
+                2,
+                [['1', '2'], ['3', '4'], ['5']],
+            ],
+            [
+                $trav([[1], [2], [3], [4], [5]]),
+                2,
+                [[[1], [2]], [[3], [4]], [[5]]],
+            ],
+            [
+                $trav([true, true, false, false]),
+                2,
+                [[true, true], [false, false]],
+            ],
+            [
+                $trav([1, 2.2, '3', [4], true, null, 'test data']),
+                3,
+                [[1, 2.2, '3'], [[4], true, null], ['test data']],
+            ],
+            [
+                $trav(['Ross', 'Rachel', 'Chandler', 'Monica', 'Joey', 'Phoebe']),
+                2,
+                [['Ross', 'Rachel'], ['Chandler', 'Monica'], ['Joey', 'Phoebe']],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider dataProviderForError
+     * @param iterable $data
+     * @param int $chunkSize
+     */
+    public function testError(iterable $data, int $chunkSize): void
+    {
+        // Given
+        try {
+            // When
+            foreach (Single::chunkwise($data, $chunkSize) as $_) {
+                break;
+            }
+            $this->fail();
+        } catch (\InvalidArgumentException $e) {
+            // Then
+            $this->assertEquals("Chunk size must be ≥ 1. Got {$chunkSize}", $e->getMessage());
+        }
+    }
```

Comment:
FYI: You can simplify this test by using PHPUnit's exception assertion mechanism and have a flat test that is much simpler.

```php
        // Then
        $this->expectException(\InvalidArgumentException::class);
        $this->expectExceptionMessage("Chunk size must be ≥ 1. Got {$chunkSize}");

        // When
        foreach (Single::chunkwise($data, $chunkSize) as $_) {
            break;
        }
```

The test will fail if no exception is thrown. I've updated the tests in develop.

Reply:
Thanks for the helpful advice!